app/models/identity_tijuana/donation.rb

```diff
@@ -0,0 +1,86 @@
+module IdentityTijuana
+  class Donation < ApplicationRecord
+    include ReadWrite
+    self.table_name = 'donations'
+    belongs_to :user
+    has_many :transactions
+
+    scope :updated_donations, -> (last_updated_at, exclude_from) {
+      where('donations.updated_at > ?', last_updated_at)
+        .and(where('donations.updated_at < ?', exclude_from))
+        .order('donations.updated_at')
+        .limit(Settings.tijuana.pull_batch_amount)
+    }
+
+    scope :updated_donations_all, -> (last_updated_at, exclude_from) {
+      where('donations.updated_at > ?', last_updated_at)
+        .and(where('donations.updated_at < ?', exclude_from))
+    }
+
+    def self.import(donation_id, sync_id)
+      donation = Donation.find(donation_id)
+      donation.import(sync_id)
+    end
+
+    def import(sync_id)
+      member = Member.find_by_external_id(:tijuana, user_id)
+      if member.present?
+        if member.ghosting_started?
+          Rails.logger.warn "Tijuana member (#{id}) is ghosted (#{existing.id}), not updating"
+        else
+          regular_donation_id = nil
+          if frequency != 'one_off'
+            regular_donation_hash = {
+              member_id: member.id,
+              # started_at: nil,
+              # ended_at: nil,
```

Comment:
These can probably be derived based on TJ's donation_upgrades table - that tracks deltas for a donation object over time.

I'm not sure what the correct model for donation changes in Id is - I feel like it doesn't get replaced with a new record when a donation upgrade happens since it has the `amount_last_changed` at column, but then that means it does lose teh formal representation of donation upgrades - which may be useful when we do upgrade fundraisers (which are pitas).

One way in which it could be inferred would be if the donations were associated with the recurring donation, but that doesn't seem to be the case?

Reply:
Donations are indeed being associated with the recurring donation in ID, via the regular_donation_id.

We don't seem to have an equivalent of the donation_upgrades table in ID, but as you've pointed out, we do have a history of donations for each regular_donation, so it can kind of be inferred.

The started_at and ended_at dates are now being populated from the information in TJ.